### PR TITLE
Feat/apikey dashboard

### PR DESCRIPTION
**Pull request by thanhdatle**
## Problem

Implement API Dashboard for user to generate APIKey, which is used to authenticate external APIs.
## Solution

A new API endpoint is introduced: 

`api/user/hasApiKey`

which returns either:

`{"message":"true"}` or `{"message":"false"}`

to indicate if user already has an API Key

If user has not generated API Key:
image

If user has generated API Key:
image

ApiKey modal is only shown once for user to copy API Key
image

The navigation header is also updated to add Dashboard and API Integration page

image

### CODE REVIEW

**Review by halfwhole — APPROVED** [review]
LGTM! Have put down a whole bunch of comments, but they're really nits that mostly have to do with minor frontend styling - works fine without!

File: src/client/apiintegration/components/ApiKeyGraphic/index.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      marginTop: theme.spacing(3),
+      alignItems: 'left',
+      [theme.breakpoints.up('md')]: {
+        marginTop: theme.spacing(6),
+      },
+    },
+    configOptionWrapper: {
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'left',
```
[the remark this answers is not in the document]
Comment (thanhdatle) [reply src/client/apiintegration/components/ApiKeyGraphic/index.tsx:33]: Done

File: src/client/apiintegration/components/ApiKeyGraphic/index.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                title="All the links you create using this API key will be linked to this account and can be found on the dashboard."
+                imageAltText="API Key help"
+              />
+            </>
+          }
+          leading={
+            <PrefixableTextField
+              value="******************"
+              onChange={() => {}}
+              placeholder="Email of link recipient"
```
[the remark this answers is not in the document]
Comment (thanhdatle) [reply src/client/apiintegration/components/ApiKeyGraphic/index.tsx:92]: Done

File: src/client/apiintegration/components/ApiKeyModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      maxWidth="sm"
+      onClose={closeApiKeyModal}
+      open={apiKeyModal}
+    >
+      <div className={classes.modalWrapper}>
+        <Typography className={classes.headerText} variant="h3" color="primary">
+          Save your API Key
+        </Typography>
+        <Typography className={classes.headerText} color="primary">
+          Please make a copy of your API key as it will be shown only once
```
[the remark this answers is not in the document]
Comment (thanhdatle) [reply src/client/apiintegration/components/ApiKeyModal/index.tsx:89]: done

File: src/client/apiintegration/index.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      <BaseLayout>
+        <ApplyAppMargins>
+          <Typography
+            className={classes.apiIntegrationHeader}
+            variant={isMinified ? 'h4' : 'h3'}
+            color="primary"
+          >
+            API Integration
+          </Typography>
+          {hasApiKey ? <ApiKeyGraphic /> : <NoApiKeyGraphic />}
```
[the remark this answers is not in the document]
Comment (thanhdatle) [reply src/client/apiintegration/index.tsx:62]: yah, this is just to make it consistent with other page. Either Graphic or Page isn't a good choice.

File: src/client/login/reducers/types.ts [hunk 5]
```diff
@@ -6,6 +6,7 @@ export type LoginState = {
   emailValidator: EmailValidatorType
   user: {
     id?: string
+    hasApiKey?: boolean
```
[the remark this answers is not in the document]
Comment (thanhdatle) [reply src/client/login/reducers/types.ts:9]: thanks for pointing out, cleaned up

File: src/client/apiintegration/components/ApiKeyGraphic/index.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  const dispatch = useDispatch()
+  return (
+    <div className={classes.root}>
+      <Typography variant="body1" className={classes.apiKeyInfoText}>
+        After generating your API key, please make a copy of it immediately as
+        it will only be shown once. Upon leaving or refreshing this page, the
+        key will be hidden.
+        <br />
+        Refer to our{' '}
+        <a href="https://go.gov.sg/go-api">
```
[the remark this answers is not in the document]
Comment (thanhdatle) [reply src/client/apiintegration/components/ApiKeyGraphic/index.tsx:72]: Good idea to move the apiDoc to gitbook

**Review by halfwhole — APPROVED** [review]

**Pull request merged**